packages/gatsby-theme-docs/gatsby-node.js

```diff
@@ -43,11 +43,13 @@ exports.onPreBootstrap = (gatsbyApi, themeOptions) => {
 exports.createResolvers = ({ createResolvers }) => {
   const resolvers = {
     SiteSiteMetadata: {
-      // this field is needed by 'gatsby-plugin-feed' plugin
+      // this field is needed by plugins needing an absolute production site URL, e.g. the'gatsby-plugin-feed' plugin
       siteUrl: {
         type: 'String',
-        resolve(source, args, context) {
-          const site = context.nodeModel.getAllNodes({ type: 'Site' })[0];
+        resolve: async (source, args, context) => {
+          const site = await context.nodeModel.findOne({
+            query: { type: 'Site' },
+          });
```

Comment:
@nkuehn FYI for the `async` resolver

Reply:
Hmm looking into this error:

```
➤ YN0000: Cannot read property 'getFields' of undefined
➤ YN0000: 
➤ YN0000: GraphQL request:7:17
➤ YN0000: 6 |                 description
➤ YN0000: 7 |                 siteUrl
➤ YN0000:   |                 ^
➤ YN0000: 8 |                 site_url: siteUrl, Cannot read property 'getFields' of undefined
➤ YN0000: 
➤ YN0000: GraphQL request:8:17
➤ YN0000: 7 |                 siteUrl
➤ YN0000: 8 |                 site_url: siteUrl
➤ YN0000:   |                 ^
➤ YN0000: 9 |               }
➤ YN0000: 
➤ YN0000: 
➤ YN0000:   Error: Cannot read property 'getFields' of undefined
➤ YN0000:   GraphQL request:7:17
➤ YN0000:   6 |                 description
➤ YN0000:   7 |                 siteUrl
➤ YN0000:     |                 ^
➤ YN0000:   8 |                 site_url: siteUrl, Cannot read property 'getFields' of und  efined
➤ YN0000:   GraphQL request:8:17
➤ YN0000:   7 |                 siteUrl
➤ YN0000:   8 |                 site_url: siteUrl
➤ YN0000:     |                 ^
➤ YN0000:   9 |               }
```

Reply:
Fixed: https://github.com/commercetools/commercetools-docs-kit/pull/1139/commits/ec27612bf0035337d3e650d2d781f3a027ae6ccf

The function options were wrong. Instead of `{ query: { type: 'Site' } }` it should be `{ type: 'Site' }`

Reply:
Thanks for spotting, the shorthand seems to be safer to use than the documented full API:  https://www.gatsbyjs.com/docs/reference/graphql-data-layer/node-model/#findOne